IOS-4891 IOS-4391 Add change BlockBook configuration

skibinalexander opened the pull request:
Нужно было переделать авторизацию для GetBlock 

https://getblock.io/docs/get-started/auth-with-access-token/

Так как протокол был один для NowNodes и GetBlock, немного переделал интерфейс общий

skibinalexander committed:
IOS-4891 IOS-4391 Add change BlockBook configuration

skibinalexander committed:
IOS-4891 Update provider changes for non nowodes changes

skibinalexander committed:
IOS-4891 Add implementation go.getblock.io

skibinalexander committed:
IOS-4891 Merge branch 'develop' into IOS-4891-update-method-auth-for-getblock

skibinalexander committed:
IOS-4891 Update data structure and rename

skibinalexander commented on BlockchainSdk/Common/BlockchainSdkConfig.swift:
```diff
@@ -110,4 +110,53 @@ public extension BlockchainSdkConfig {
         }
     }
+    
+    struct GetBlockCredentials {
```
@tureck1y посмотри пожалуйста, самому не очень нравится что появляются методы получения кредов для GetBlock но как сделать и где , более понятно и удобно пока не вижу

tureck1y replied:
только бы вложенные типы Credential и TypeValue положить в extension, чтобы проще читать было

tureck1y commented on BlockchainSdk/BlockchainService/BlockBook/Config/Implementations/GetBlockBlockBookConfig.swift:
```diff
@@ -8,42 +8,42 @@
     func node(for blockchain: Blockchain) -> BlockBookNode {
-        let currencySymbolPrefix = blockchain.currencySymbol.lowercased()
+        let rpcApiKeyValue = credentialsConfig.credential(for: blockchain, at: .jsonRpc)
+        let blockBookApiKeyValue = credentialsConfig.credential(for: blockchain, at: .blockBook)
```
может сделать restNodeApiKeyValue? Чтобы билось с rpc?

tureck1y commented on BlockchainSdk/BlockchainService/BlockBook/Config/Implementations/GetBlockBlockBookConfig.swift:
```diff
@@ -8,42 +8,42 @@
-    var apiKeyName: String {
-        return Constants.xApiKeyHeaderName
-    }
+    var apiKeyValue: String { "" }
+    var apiKeyName: String { "" }
```
а от эт ого можно будет избавиться?

tureck1y commented on BlockchainSdk/Common/Blockchain.swift:
```diff
@@ -548,7 +548,7 @@ extension Blockchain {
         case .gnosis:
             return [
-                URL(string: "https://gno.getblock.io/mainnet?api_key=\(getBlockApiKey)")!,
+                getGetBlockProvider(for: .gnosis),
```
makeGetBlockProvider ?

tureck1y commented on BlockchainSdk/Common/BlockchainSdkConfig.swift:
```diff
@@ -110,4 +110,53 @@ public extension BlockchainSdkConfig {
+extension BlockchainSdkConfig.GetBlockCredentials {
+    func credential(
+        for blockchain: Blockchain,
+        at type: Credential.TypeValue
```
я бы тут и ниже убрал at. Просто type

skibinalexander committed:
Merge branch 'develop' into IOS-4891-update-method-auth-for-getblock

skibinalexander committed:
IOS-4891 Merge branch 'develop' into IOS-4891-update-method-auth-for-getblock

# Conflicts:
#	BlockchainSdk/Common/WalletAssembly/NEARWalletAssembly.swift
#	BlockchainSdk/WalletManagers/Cosmos/CosmosChain.swift

skibinalexander committed:
IOS-4891 Fix review comments

skibinalexander committed:
IOS-4891 Fix rename makeGetBlockJsonRpcProvider

tureck1y reviewed (commented):
смотрю

skibinalexander committed:
Merge branch 'develop' into IOS-4891-update-method-auth-for-getblock

Balashov152 commented on BlockchainSdk/Common/Blockchain.swift:
```diff
@@ -444,7 +444,7 @@ extension Blockchain {
                 return [
                     URL(string: "https://eth.nownodes.io/\(nowNodesApiKey)")!,
-                    URL(string: "https://eth.getblock.io/mainnet?api_key=\(getBlockApiKey)")!,
+                    makeGetBlockJsonRpcProvider(for: .ethereum(testnet: false)),
```
может лучше через `self`

```suggestion
                    makeGetBlockJsonRpcProvider(for: self),
```

Balashov152 commented on BlockchainSdk/Common/Blockchain.swift:
```diff
@@ -634,6 +634,18 @@ extension Blockchain {
+        
+        // MARK: - Private Implementation
+        
+        func makeGetBlockJsonRpcProvider(for blockchain: Blockchain) -> URL {
```
можно прям тут тогда убрать параметр, и просто возвращать от `self`

Balashov152 commented on BlockchainSdk/Common/BlockchainSdkConfig.swift:
```diff
@@ -110,4 +110,50 @@ public extension BlockchainSdkConfig {
+        return credential?.value ?? ""
+    }
+    
+    func credentials(type: TypeValue) -> [Blockchain: String] {
+        var resultJsonRpcApiKeys = [Blockchain: String]()
+        
+        credentials
+            .filter { $0.type == type }
+            .forEach { resultJsonRpcApiKeys[$0.blockchain] = $0.value }
+        
+        return resultJsonRpcApiKeys
+    }
```
Можно через reduce собрать 

```suggestion
    func credentials(type: TypeValue) -> [Blockchain: String] {
        credentials
            .filter { $0.type == type }
            .reduce(into: [:]) { $0[$1.blockchain] = $1.value }
    }
```

skibinalexander replied:
> credentials
>             .filter { $0.type == type }
>             .reduce(into: [:]) { $0[$1.blockchain] = $1.value }

Ага, спс, так получше будет

skibinalexander committed:
IOS-4891 Update keys type

skibinalexander committed:
IOS-4891 Fix review comments
